server/svix-server/src/queue/rabbitmq.rs

```diff
@@ -0,0 +1,269 @@
+use chrono::Utc;
+use futures::StreamExt;
+use lapin::{
+    options::{
+        BasicConsumeOptions, BasicPublishOptions, BasicQosOptions, ExchangeDeclareOptions,
+        QueueBindOptions, QueueDeclareOptions,
+    },
+    types::{AMQPValue, FieldTable},
+    BasicProperties, ConnectionProperties,
+};
+use svix_ksuid::{KsuidLike, KsuidMs};
+
+use crate::{ctx, err_generic, err_queue, error::Result};
+use std::{sync::Arc, time::Duration};
+
+use super::{
+    Acker, QueueTask, TaskQueueConsumer, TaskQueueDelivery, TaskQueueProducer, TaskQueueReceive,
+    TaskQueueSend,
+};
+
+#[derive(Clone)]
+pub struct Producer(Arc<ProducerInner>);
+
+struct ProducerInner {
+    exchange_name: String,
+    queue_name: String,
+    channel: lapin::Channel,
+}
+
+pub struct Consumer {
+    consumer: lapin::Consumer,
+}
+
+pub async fn new_pair(
+    dsn: &str,
+    queue_name: String,
+) -> Result<(TaskQueueProducer, TaskQueueConsumer)> {
+    let conn = ctx!(lapin::Connection::connect(dsn, ConnectionProperties::default()).await)?;
+    let producer_chan = ctx!(conn.create_channel().await)?;
+    let consumer_chan = ctx!(conn.create_channel().await)?;
+
+    let exchange_name = ctx!(declare_delayed_message_exchange(&producer_chan).await)?;
+    ctx!(declare_bound_queue(&queue_name, &exchange_name, &producer_chan).await)?;
+
+    // With prefetch_size, there's a tradeoff.
+    // More prefetched messages drains RabbitMQ Queues faster, but also consumes more worker memory.
+    // Additionally, too many prefetched messages can starve other workers, and hurt total throughput.
+    // Leaving as 1 for now, to prevent starvation.
+    let prefetch_size = 1;
```

Comment:
Does prefetch define a minimum as well as a maximum?

Reply:
... either way what I'm getting at though is that I think this should be configurable or (gasp) we should modify `receive_all`  to accept an optional max batch-size (though not sure if it's possible to specify at receive-time w/ rabbit)

Reply:
I didn't think there was? Gabriel may have seen the docs on this more recently than I. 

If you had a minimum, you'd usually need some other heuristic to prevent _"N where N  < PREFETCH_MIN"_ messages sitting there for ages with nobody electing to pick them up during slower periods.

Reply:
It's a maximum. https://www.rabbitmq.com/amqp-0-9-1-reference.html#basic.qos.prefetch-size

It's not necessarily related to how many values `receive_all` returns, it's just a tuning variable (basically trading RabbitMQ queue memory for consumer client's memory, with the other caveats mentioned in the comment.

I think having this as a configurable variable makes sense though. I'll make that change